src/modules/hip/kernel/resize_mirror_normalize.hpp

```diff
@@ -0,0 +1,369 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+
+__device__ void rmn_hip_compute(uchar *srcPtr, d_float8 *pix_f8, d_float8 *rmnParams_f8)
+{
+    pix_f8->f4[0] = rpp_hip_pixel_check_0to255((pix_f8->f4[0] - rmnParams_f8->f4[0]) * rmnParams_f8->f4[1]);
+    pix_f8->f4[1] = rpp_hip_pixel_check_0to255((pix_f8->f4[1] - rmnParams_f8->f4[0]) * rmnParams_f8->f4[1]);
+}
+
+__device__ void rmn_hip_compute(float *srcPtr, d_float8 *pix_f8, d_float8 *rmnParams_f8)
+{
+    pix_f8->f4[0] = rpp_hip_pixel_check_0to1((pix_f8->f4[0] - rmnParams_f8->f4[0] * (float4) ONE_OVER_255) * rmnParams_f8->f4[1]);
+    pix_f8->f4[1] = rpp_hip_pixel_check_0to1((pix_f8->f4[1] - rmnParams_f8->f4[0] * (float4) ONE_OVER_255) * rmnParams_f8->f4[1]);
+}
+
+__device__ void rmn_hip_compute(schar *srcPtr, d_float8 *pix_f8, d_float8 *rmnParams_f8)
+{
+    pix_f8->f4[0] = rpp_hip_pixel_check_0to255(((pix_f8->f4[0] + (float4)128) - rmnParams_f8->f4[0]) * rmnParams_f8->f4[1]) - (float4)128;
+    pix_f8->f4[1] = rpp_hip_pixel_check_0to255(((pix_f8->f4[1] + (float4)128) - rmnParams_f8->f4[0]) * rmnParams_f8->f4[1]) - (float4)128;
+}
+
+__device__ void rmn_hip_compute(half *srcPtr, d_float8 *pix_f8, d_float8 *rmnParams_f8)
+{
+    pix_f8->f4[0] = rpp_hip_pixel_check_0to1((pix_f8->f4[0] - rmnParams_f8->f4[0] * (float4) ONE_OVER_255) * rmnParams_f8->f4[1]);
+    pix_f8->f4[1] = rpp_hip_pixel_check_0to1((pix_f8->f4[1] - rmnParams_f8->f4[0] * (float4) ONE_OVER_255) * rmnParams_f8->f4[1]);
+}
+
+__device__ void resize_mirror_normalize_roi_and_srclocs_hip_compute(int4 *srcRoiPtr_i4, uint2 *dstDimsWH, int id_x, int id_y, d_float16 *locSrc_f16)
+{
+    float wRatio = float(srcRoiPtr_i4->z) / (float)dstDimsWH->x;
+    float hRatio = float(srcRoiPtr_i4->w) / (float)dstDimsWH->y;
+    float wOffset = (wRatio - 1) * 0.5f;
+    float hOffset = (hRatio - 1) * 0.5f;
+
+    d_float8 increment_f8, locDst_f8x, locDst_f8y;
+    increment_f8.f4[0] = make_float4(0.0f, 1.0f, 2.0f, 3.0f);
+    increment_f8.f4[1] = make_float4(4.0f, 5.0f, 6.0f, 7.0f);
+    locDst_f8x.f4[0] = float4(id_x) + increment_f8.f4[0];
+    locDst_f8x.f4[1] = float4(id_x) + increment_f8.f4[1];
+    locDst_f8y.f4[0] = float4(id_y);
+    locDst_f8y.f4[1] = float4(id_y);
+
+    locSrc_f16->f8[0].f4[0] = (locDst_f8x.f4[0] * float4(wRatio)) + float4(wOffset);  // Compute First 4 locSrcX
+    locSrc_f16->f8[0].f4[1] = (locDst_f8x.f4[1] * float4(wRatio)) + float4(wOffset);  // Compute Next 4 locSrcX
+    locSrc_f16->f8[1].f4[0] = (locDst_f8y.f4[0] * float4(hRatio)) + float4(hOffset);  // Compute First 4 locSrcY
+    locSrc_f16->f8[1].f4[1] = (locDst_f8y.f4[1] * float4(hRatio)) + float4(hOffset);  // Compute Next 4 locSrcY
+}
+
+__device__ void resize_mirror_normalize_roi_and_srclocs_hip_compute_mirror(int4 *srcRoiPtr_i4, uint2 *dstDimsWH, int id_x, int id_y, d_float16 *locSrc_f16)
```

Comment:
Yeah x,y components are not needed only z,w is needed
But it will be used in later stages for interpolation functions
https://github.com/sampath1117/rpp/blob/sr/opt_rmn/src/modules/hip/kernel/resize_mirror_normalize.hpp#L108